### PR TITLE
DefaultBrowserContext tests

### LINKED ISSUE #8: Add defaultbrowsercontext.spec.js tests
Based on this tag https://github.com/kblok/playwright/tree/first-snapshot. We don't need to get the tests green. We only need to incoporate the code. Let's also check if we can get code from Puppeteer

### PR DESCRIPTION
**Pull request by Meir017**
closes #8

### CODE REVIEW

File: src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        [Fact]
+        public async Task PageSetCookiesAsyncShouldWork()
+        {
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            await Page.BrowserContext.SetCookiesAsync(new SetNetworkCookieParam
+            {
+                Name = "username",
+                Value = "John Doe"
+            });
+
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs]: Missing eval check here
```
expect(await page.evaluate(() => document.cookie)).toBe('username=John Doe');
```

File: src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            Assert.Equal(16, cookie.Size);
+            Assert.False(cookie.HttpOnly);
+            Assert.False(cookie.Secure);
+            Assert.True(cookie.Session);
+        }
+
+        /// <playwright-file>defaultbrowsercontext.spec.js</playwright-file>
+        /// <playwright-file>context.clearCookies() should work</playwright-file>
+        [Fact]
+        public async Task PageDeleteCookieAsyncShouldWork()
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs]:
```suggestion
        public async Task ContextClearCookiesAsyncShouldWork()
```

File: src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            Assert.Equal(16, cookie.Size);
+            Assert.False(cookie.HttpOnly);
+            Assert.False(cookie.Secure);
+            Assert.True(cookie.Session);
+        }
+
+        /// <playwright-file>defaultbrowsercontext.spec.js</playwright-file>
+        /// <playwright-test>context.setCookies() should work</playwright-test>
+        [Fact]
+        public async Task PageSetCookiesAsyncShouldWork()
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs]:
```suggestion
        public async Task ContextSetCookiesAsyncShouldWork()
```

File: src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    public class DefaultBrowserContextTests : PlaywrightSharpPageBaseTest
+    {
+        internal DefaultBrowserContextTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <playwright-file>defaultbrowsercontext.spec.js</playwright-file>
+        /// <playwright-test>page.cookies() should work</playwright-test>
+        [Fact]
+        public async Task PageGetCookiesAsyncShouldWork()
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs]:
```suggestion
        public async Task ContextGetCookiesAsyncShouldWork()
```

File: src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    Value = "1"
+                },
+                new SetNetworkCookieParam
+                {
+                    Name = "cookie2",
+                    Value = "2"
+                });
+
+            Assert.Equal("cookie1=1; cookie2=2", await Page.EvaluateExpressionAsync<string>("document.cookie"));
+            await Page.BrowserContext.DeleteCookiesAsync(new SetNetworkCookieParam
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/DefaultBrowserContextTests.cs]: This test doesn't match with the playwright test.

File: src/PlaywrightSharp.Abstractions/NavigationOptions.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    /// <summary>
+    /// Navigation options used by <see TODOcref="Page.WaitForNavigationAsync(NavigationOptions)"/> and <see TODOcref="Page.SetContentAsync(string, NavigationOptions)"/>
+    /// </summary>
+    public class NavigationOptions
+    {
+        /// <summary>
+        /// Maximum navigation time in milliseconds, defaults to 30 seconds, pass <c>0</c> to disable timeout. 
+        /// </summary>
+        /// <remarks>
+        /// The default value can be changed by setting the <see TODOcref="Page.DefaultNavigationTimeout"/> or <see TODOcref="Page.DefaultTimeout"/> property.
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/NavigationOptions.cs]: Let's create those page properties.

File: src/PlaywrightSharp.Abstractions/SetNetworkCookieParam.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
+﻿namespace PlaywrightSharp
+{
+    /// <summary>
+    /// Cookie data.
+    /// </summary>
+    /// <seealso TODOcref="Page.SetContentAsync(string, NavigationOptions)"/>
+    /// <seealso TODOcref="Page.DeleteCookieAsync(SetNetworkCookieParam[])"/>
+    /// <seealso TODOcref="Page.GetCookiesAsync(string[])"/>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/SetNetworkCookieParam.cs]: Let's remove this TODOs. You should mention browser functions

File: src/PlaywrightSharp.Abstractions/SetNetworkCookieParam.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    public class SetNetworkCookieParam
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/SetNetworkCookieParam.cs]: Let's call this NetworkCookie

File: src/PlaywrightSharp.Abstractions/IBrowserContext.cs [hunk 9]
```diff
@@ -27,11 +26,37 @@ public interface IBrowserContext
         /// </summary>
         /// <returns>A <see cref="Task{IPage}"/> that completes when a new <see cref="IPage"/> is created</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IBrowserContext.cs]:
```suggestion
        /// <returns>A <see cref="Task{IPage}"/> that completes when a new <see cref="IPage"/> is created</returns>, yielding the new <see cref="IPage"/>.
```

File: src/PlaywrightSharp.Abstractions/IBrowserContext.cs [hunk 10]
```diff
@@ -27,11 +26,37 @@ public interface IBrowserContext
...
         /// </summary>
         /// <remarks>NOTE only incognito browser contexts can be closed.</remarks>
         /// <returns>A <see cref="Task"/> that completes when the browser context is closed</returns>
         Task CloseAsync();
+
+        /// <summary>
+        /// Returns the context's cookies
+        /// </summary>
+        /// <param name="urls">Url's to return cookies for</param>
+        /// <returns>Array of cookies</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IBrowserContext.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the cookies are sent by the browser, yielding a <see cref="t:NetworkCookie[]"/></returns>
```

File: src/PlaywrightSharp.Abstractions/IBrowserContext.cs [hunk 11]
```diff
@@ -27,11 +26,37 @@ public interface IBrowserContext
...
+        /// If no URLs are specified, this method returns cookies for the current page URL.
+        /// If URLs are specified, only cookies for those URLs are returned.
+        /// </remarks>
+        Task<SetNetworkCookieParam[]> GetCookiesAsync(params string[] urls);
+
+        /// <summary>
+        /// Clears all of the current cookies and then sets the cookies for the context
+        /// </summary>
+        /// <param name="cookies">Cookies to set</param>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IBrowserContext.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the cookies are set</returns>
```

File: src/PlaywrightSharp.Abstractions/IBrowserContext.cs [hunk 12]
```diff
@@ -27,11 +26,37 @@ public interface IBrowserContext
...
+        Task SetCookiesAsync(params SetNetworkCookieParam[] cookies);
+
+        /// <summary>
+        /// Deletes cookies from the context
+        /// </summary>
+        /// <param name="cookies">Cookies to delete</param>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IBrowserContext.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the cookies are deleted.</returns>
```

File: src/PlaywrightSharp.Abstractions/SetNetworkCookieParam.cs (hunk 8 above)
Comment (Meir017) [reply src/PlaywrightSharp.Abstractions/SetNetworkCookieParam.cs]: in playwright there are 2 interfaces - `NetworkCookie` & `SetNetworkCookieParam` - https://github.com/kblok/playwright/blob/master/src/network.ts

**Comment by Meir017** [comment]
@kblok ready

File: src/PlaywrightSharp.Abstractions/IPage.cs [hunk 13]
```diff
@@ -115,8 +154,53 @@ public interface IPage
         /// Sets the HTML markup to the page
         /// </summary>
         /// <param name="html">HTML markup to assign to the page.</param>
-        /// <returns>A <see cref="Task"/> that completes when the javascript code executing injected the HTML finishes</returns>
-        Task SetContentAsync(string html);
+        /// <param name="options">The navigations options</param>
+        /// <returns>Task.</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IPage.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the javascript code executing injected the HTML finishes</returns>
```

File: src/PlaywrightSharp.Abstractions/IBrowserContext.cs [hunk 14]
```diff
@@ -42,5 +42,50 @@ public interface IBrowserContext
         /// A <see cref="Task"/> that completes when get browser context got all the pages, yielding the pages inside that browser context.
         /// </returns>
         Task<IPage[]> GetPagesAsync();
+
+        /// <summary>
+        /// Sets the HTML markup to the browser context
+        /// </summary>
+        /// <param name="html">HTML markup to assign to the page.</param>
+        /// <param name="options">The navigations options</param>
+        /// <returns>Task.</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IBrowserContext.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the javascript code executing injected the HTML finishes</returns>
```

File: src/PlaywrightSharp.Abstractions/IFrame.cs [hunk 15]
```diff
@@ -57,6 +57,16 @@ public interface IFrame
         /// </summary>
         string Url { get; }
 
+        /// <summary>
+        /// Sets the HTML markup to the frame
+        /// </summary>
+        /// <param name="html">HTML markup to assign to the page.</param>
+        /// <param name="options">The navigations options</param>
+        /// <returns>Task.</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IFrame.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the javascript code executing injected the HTML finishes</returns>
```

File: src/PlaywrightSharp.Abstractions/IFrame.cs [hunk 16]
```diff
@@ -1,94 +1,104 @@
 ﻿using System.Text.Json;
-using System.Threading.Tasks;
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IFrame.cs]: What happened here? I pushed an .editorconfig Is your IDE using that?

File: src/PlaywrightSharp.Abstractions/IPage.cs [hunk 17]
```diff
@@ -115,8 +154,53 @@ public interface IPage
...
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that completes when the cookies are cleared.</returns>
+        /// <seealso cref="IBrowserContext.ClearCookiesAsync"/>
+        Task ClearCookiesAsync();
+
+        /// <summary>
+        /// Sets extra HTTP headers that will be sent with every request the page initiates
+        /// </summary>
+        /// <param name="headers">Additional http headers to be sent with every request</param>
+        /// <returns>Task</returns>
```
Comment (kblok) [inline src/PlaywrightSharp.Abstractions/IPage.cs]:
```suggestion
        /// <returns>A <see cref="Task"/> that completes when the headers are set.</returns>
```

File: src/PlaywrightSharp.Abstractions/IFrame.cs (hunk 16 above)
Comment (Meir017) [reply src/PlaywrightSharp.Abstractions/IFrame.cs]: hmm, not sure what happened here

Comment (kblok) [reply src/PlaywrightSharp.Abstractions/IFrame.cs]: If we fix that this would be ready to go 🚀

**Pull request merged**